### PR TITLE
ENH add a script to annotate NiMARE with topics

### PR DESCRIPTION
**Pull request by JulioAPeraza**
add a script to scrape local html files from https://neurosynth.org/ and to annotate the NiMARE version of the Neurosynth dataset.

### CODE REVIEW

File: annotate_neurosynth.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    download(out_dir, unpack=True)
+
+# Convert Neurosynth database files to NiMARE Dataset
+dset = nimare.io.convert_neurosynth_to_dataset(
+    os.path.join(out_dir, 'database.txt'),
+    os.path.join(out_dir, 'features.txt'))
+
+dset.save(os.path.join(out_dir, 'neurosynth_dataset.pkl.gz'))
+
+"""
```
Comment (tsalo) [inline annotate_neurosynth.py]: Can you move this docstring up to the top of the script?

File: annotate_neurosynth.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            # title.append(study.text)
+
+    found_ids = np.isin(dset.ids, list(nimare_ids_weight.keys()))
+    #found_ids = dset.metadata['title'].isin(title)
+    ids_colum = np.zeros(len(dset.ids))
+    topic_weights = [nimare_ids_weight[id] for id in dset.ids[found_ids]]
+    ids_colum[found_ids] = topic_weights
+
+    nonzero = ids_colum[ids_colum != 0]
+    # doble check that all the studies are in the dataset and match the number of studies reported on https://neurosynth.org/analyses/topics/v5-topics-200/
```
Comment (tsalo) [inline annotate_neurosynth.py]: split comment into two lines

File: annotate_neurosynth.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    - Neurosynth_v5topic200_01_topics.htm
+    - ...
+ Download the pages of each topic (https://neurosynth.org/analyses/topics/v5-topics-200/0) as:
+    - Neurosynth_v5topic200_topic000_00.htm
+    - Neurosynth_v5topic200_topic000_01.htm
+    - ...
+"""
+#dset = Dataset.load(os.path.join(out_dir, 'neurosynth_dataset.pkl.gz'))
+
+topics = 'v5topic200'
```
Comment (tsalo) [inline annotate_neurosynth.py]: Please make constants capitalized to match PEP8 (e.g., `TOPICS`).

File: annotate_neurosynth.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        studies_table = soup_topic.find('div', class_='row').find('div', class_='row').find(
+            'div', class_='col-md-10 content').find('div', class_='tab-content').find('div', id='studies').table.tbody
+        studies = studies_table.find_all('a')
+        weights = studies_table.find_all('td', class_='sorting_1')
+
+        for i, study in enumerate(studies):
+            expid = '1'
+            pid = study['href'].split('/')[4]
+            nimare_ids_weight["{0}-{1}".format(pid, expid)] = weights[i].text
+            # title.append(study.text)
```
Comment (tsalo) [inline annotate_neurosynth.py]: Trim old, commented-out code.

File: annotate_neurosynth.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
+import os
+from neurosynth.base.dataset import download
+import nimare
+from bs4 import BeautifulSoup
+import numpy as np
+import glob
+#from nimare.dataset import Dataset
```
Comment (tsalo) [inline annotate_neurosynth.py]: Could you `isort` this?

**Review by tsalo — CHANGES_REQUESTED** [review]
Just some small changes requested.

Thanks so much for this!

**Comment by JulioAPeraza** [comment]
Thanks!

I did a final test and I got this warning:
`<frozen importlib._bootstrap>:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192 from C header, got 216 from PyObject`

**Comment by tsalo** [comment]
Is that the full traceback?

**Comment by JulioAPeraza** [comment]
Yes. The function still working, but now I get that warning. Maybe I broke my python environment with the latest numpy update.

**Comment by tsalo** [comment]
Weird...

Has this been tested against NiMARE 0.0.3? I'd like to update the datasets in https://osf.io/b2ycs/.

**Comment by JulioAPeraza** [comment]
I am testing on NiMARE: `0.0.3+31.g530e652`

**Comment by tsalo** [comment]
It'll probably work on 0.0.3, although the current dev version (which is `0.0.3+[number of commits]`) has substantial changes to the Dataset object. I'll merge as-is and work from there. Thank you!

**Pull request merged**